examples/dev_example/docs/build.jl

```diff
@@ -0,0 +1,28 @@
+import Lexicon.Elements: config, section, document, page
+
+import Lexicon.Render: save, markdown
+
+import Docile: Cache, Collector
+
+out = document("TestDocumentation",
+    section("Introduction",
+        page("index", "index.md"),
+        ),
+    section("Docile-API",
+        page("Cache", config(style_obj="###"),
+            "# Cache",
+            "## Macros",
+            "Some intermediate text describing more about macros.",
+            (Cache, config(objfilter=[:macro])),
+            "## Methods",
+            (Cache, config(objfilter=[:method])),
+            ),
+        page("Collector", config(objfilter=[:method, :macro], style_obj="###"),
+            "# Collector",
+            "## Methods & Macros",
+            Collector,
+            ),
+        ),
+    )
```

Comment:
This is looking much nicer.

Instead of requiring ``Module``s to be separated with tuples for applying different `config` could we just allow nested `section` inside of `page`? The less syntax things to have to remember the better I think.

The first argument to each `page` node appears to be a title, yes? And then it's kind of repeated again as the third argument. Reasons for this? Do we need the repetition?